libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -274,8 +314,12 @@ export class PackageExtractor {
     // Create a new state for each run
     const state: IExtractorState = {
       foldersToCopy: new Set(),
+      packageJsonByPath: new Map(),
       projectConfigurationsByName: new Map(projectConfigurations.map((p) => [p.projectName, p])),
       projectConfigurationsByPath: new Map(projectConfigurations.map((p) => [p.projectFolder, p])),
+      dependencyConfigurationsByNameAndVersion: new Map(
+        dependenciesConfigurations.map((p) => [`${p.dependencyName}@${p.dependencyVersion}`, p])
```

Comment:
If we turn the version into a version range, this will likely have to turn into a `Map<string, IExtractorDependencyConfiguration[]>` which is name -> config, and then cycle through the included configs to check the ranges with semver.

Reply:
LGTM.
But as the Map structure has changed, it can't be easily convert from Array. So I add several lines to covert origin Array to Map.